### ISSUE TITLE
push_redirect inside a LiveComponent crashes the new LiveView after a few seconds

### ISSUE THREAD

**Issue by nicolasblanco** [phoenixframework/phoenix_live_view#1293]
### Environment

* Elixir version (elixir -v): 1.11.2
* Phoenix version (mix deps): 1.5.7
* Phoenix LiveView version (mix deps): 0.15.1
* NodeJS version (node -v): v15.5.0
* NPM version (npm -v): 7.3.0
* Operating system: OS X Big Sur
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome, Safari
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual behavior

Create a new application with LiveView included:
```
mix phx.new my_app --live
```

Create a simple component that is redirecting to the same LiveView (`/`) with `push_redirect`:

```
defmodule MyAppWeb.RedirectComponent do
  use MyAppWeb, :live_component

  def render(assigns) do
    ~L"""
    <div class="hero">
      <a phx-click="redirect" phx-target="<%= @myself %>">Test</a>
    </div>
    """
  end

  def handle_event("redirect", _, socket) do
    {:noreply, push_redirect(socket, to: "/")}
  end
end
```

Display it in `page_live.html.leex`:

```
<%= live_component @socket, MyAppWeb.RedirectComponent, id: "redirect" %>
```

Click on the link. The LV is reloaded.
Wait 10 seconds (more or less), the LV crashes and the page is reloaded.

When activating the debugger I'm getting something like that:

<img width="1182" alt="Screenshot 2020-12-27 at 05 57 29" src="https://user-images.githubusercontent.com/12682/103164448-f3fdb100-4813-11eb-983c-9e5782e34e88.png">

The first arrow is when the `push_redirect` is executed, we can see the new Liveview being mounted. Then for some reason a few dozen seconds later I'm getting a reference to the previously destroyed LV and then the new LV crashes and the page reloads.

### Expected behavior

The new LV being created after the `push_redirect` should not crash.

Thanks for your help 👍🏻

**Comment by mveytsman** [comment]
I can confirm this happens in master -- looks like the `pushWithReply` for the click is timing out.

**Comment by mveytsman** [comment]
https://github.com/phoenixframework/phoenix_live_view/blob/c263db32f446e57275f62f507f4995fef59c1975/lib/phoenix_live_view/channel.ex#L459-L485

I think the call to `handle_redirect` in here should be passed the `ref` instead of nil. This fixes the issue, but breaks a test for live patching from within components and probably has some implications about flashes that I don't understand yet. TBH this is at the edge of my understanding of the channel internals right now :)

**Comment by nicolasblanco** [comment]
This bug can be reproduced by simply using the `phx.gen.live` generator and a simple scaffold because the generator is using `push_redirect` inside the `FormComponent`.

Just generate a scaffold, create a new item, wait 10 seconds and the LV crashes.

<img width="962" alt="Screenshot 2021-01-01 at 20 53 45" src="https://user-images.githubusercontent.com/12682/103444699-a352e180-4c73-11eb-93dc-b558c29c8a83.png">

**Comment by nicolasblanco** [comment]
> I can confirm this happens in master -- looks like the `pushWithReply` for the click is timing out. This seems related: [2ece589](https://github.com/phoenixframework/phoenix_live_view/commit/2ece58925de82d2bb28f3351399b7d77c24f5c8c)
> 
> Looks like the check we made there isn't doing the job when you're redirecting back to the same view -- joinCount stays the same, but we destroyed and recreated the component.

I confirm that this issue has been introduced in `0.15.1`. I cannot reproduce it in `0.15.0`.

**Comment by chrismccord** [comment]
Fixed and released in 0.15.3. Thanks!
